griffon/commands/entities.py

```diff
@@ -46,64 +222,155 @@ def flaws(ctx):
+
+    session = OSIDBService.create_session()
+    data = session.flaws.retrieve(flaw_id, **params)
+    return cprint(data, ctx=ctx)
+
+
+@flaws.command(name="update")
 @click.option(
-    "--embargoed",
-    "is_embargoed",
-    is_flag=True,
-    help="Include embargoed flaws (requires access).",
+    "--id",
+    "flaw_id",
+    help="Flaw CVE-ID or UUID.",
+    required=True,
+)
+@request_body_options(
+    endpoint_module=osidb_api_v1_flaws_update, exclude=["uuid", "trackers", "created_dt"]
 )
-@click.option("--major-incident", "is_major_incident", is_flag=True, help="Only major incidents.")
 @click.pass_context
 @progress_bar
-def list_flaws(ctx, flaw_state, resolution, impact, is_embargoed, is_major_incident):
-    if not flaw_state and not resolution and not impact:
-        click.echo(ctx.get_help())
-        exit(0)
+def update_flaw(ctx, flaw_id, **params):
+    request_body_type = getattr(osidb_api_v1_flaws_update, "REQUEST_BODY_TYPE", None)
+    if request_body_type is None:
+        raise click.ClickException(
+            "No request body template for Flaw update. "
+            "Is correct version of osidb-bindings installed?"
+        )
+
+    fields = filter_request_fields(
+        request_body_type.get_fields(), exclude=["uuid", "trackers", "created_dt"]
+    )
+    params = multivalue_params_to_csv(params)
+
     session = OSIDBService.create_session()
-    conditions = default_conditions
-    if flaw_state:
-        conditions["state"] = flaw_state
-    if resolution:
-        conditions["resolution"] = resolution
-    if impact:
-        conditions["impact"] = impact
-    data = session.flaws.retrieve_list(**conditions).results
+
+    try:
+        data = session.flaws.retrieve(id=flaw_id, include_fields=",".join(fields))
+    except Exception as e:
+        if ctx.obj["VERBOSE"]:
+            console.log(e, e.response.json())
+        raise click.ClickException(
+            f"Failed to fetch Flaw with ID '{flaw_id}'. "
+            "Flaw either does not exist or you have insufficient permissions. "
+            "Consider running griffon with -v option for verbose error log."
+        )
+
+    data = data.to_dict()
+    # remove status data from OSIDB server
+    [data.pop(key) for key in ["dt", "env", "revision", "version"]]
+    data.update((field, value) for field, value in params.items() if value is not None)
+
+    if ctx.obj["EDITOR"]:
+        data = click.edit(text=json.dumps(data, indent=4), editor=get_editor(), require_save=False)
+        data = json.loads(data)
+
+    try:
+        data = session.flaws.update(id=flaw_id, form_data=data)
+    except HTTPError as e:
+        if ctx.obj["VERBOSE"]:
+            console.log(e, e.response.json())
+        raise click.ClickException(
+            f"Failed to update Flaw with ID '{flaw_id}'. "
+            "You might have insufficient permission or you've supplied malformed data. "
+            "Consider running griffon with -v option for verbose error log."
```

Comment:
-v is normally to do with output... --debug is what I think we want to mention here.

Reply:
nm I see you have enhanced VERBOSE

Reply:
I've used that whatever level of `-v` will print the verbose error log. Do you think it should be better to do that with `--debug`? I can change.